WordPress/src/main/java/org/wordpress/android/ui/prefs/AppSettingsFragment.java

```diff
@@ -136,13 +136,13 @@ private void changeLanguage(String languageCode) {
         Configuration conf = res.getConfiguration();
         // will return conf.locale if conf is non-null, or Locale.getDefault()
         Locale currentLocale = LanguageUtils.getCurrentDeviceLanguage(WordPress.getContext());
+        Locale newLocale = WPPrefUtils.languageLocale(languageCode);
 
-        if (currentLocale.getLanguage().equals(new Locale(languageCode).getLanguage())) {
+        if (currentLocale.toString().equals(newLocale.getDisplayLanguage())) {
             return;
         }
 
-        Locale newLocale = WPPrefUtils.languageLocale(languageCode);
-        if (currentLocale.getLanguage().equals(newLocale.getLanguage())) {
+        if (Locale.getDefault().toString().equals(newLocale.toString())) {
```

Comment:
Commit b07a41b dropped `toString()` from the locale equality comparison as recommended by the [docs](https://docs.oracle.com/javase/7/docs/api/java/util/Locale.html#getLanguage()), but here `toString()` has been added back to the comparison. Was this intentional?